### PR TITLE
[MPS] Add `torch.fft.` support

### PR DESCRIPTION
**Pull request by malfet**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #119670

Increase tolerance for `ftt` ops, that warrants a further investigation as it grows larger with larger matrix dimensions (see https://github.com/pytorch/pytorch/issues/120237 )

When compiling on MacOS13, implement `+[FakeMPSGraphFFTDescriptor descriptor]` as a redispatch to a real thing.

Fixes https://github.com/pytorch/pytorch/issues/78044

### CODE REVIEW

File: aten/src/ATen/native/SpectralOpsUtils.h [hunk 1]
```diff
@@ -3,8 +3,14 @@
 #include <string>
 #include <stdexcept>
 #include <sstream>
+#include <c10/core/ScalarType.h>
+#include <c10/util/ArrayRef.h>
+#include <c10/util/Exception.h>
 #include <ATen/native/DispatchStub.h>
 
+namespace at {
+class Tensor;
```
Comment (albanD) [inline aten/src/ATen/native/SpectralOpsUtils.h:12]: Why do you need that?

**Review by albanD — COMMENTED** [review]
I'll let @kulinseth review the objc part.
The tolerance difference looks suspicious indeed. I don't know off the top of my head if FFT tolerance is supposed to scale with the input size but that sounds surprising.
If you do want to land this as-is, please make sure to open an issue about it and have a todo next to all the tolerance bumps so we can keep track of them.

File: aten/src/ATen/native/SpectralOpsUtils.h (hunk 1 above)
Comment (malfet) [reply aten/src/ATen/native/SpectralOpsUtils.h:12]: As linter wants to include SpectralOpsUtils.h first on the list, and want to fwd declare tensor. But can replace it with `#include` if needed

Comment (albanD) [reply aten/src/ATen/native/SpectralOpsUtils.h:12]: If including TensorBase is enough I would.
Otherwise the forward declaration is fine yep

**Review by kulinseth — APPROVED** [review]
Looks good.

**Review by albanD — APPROVED** [review]
SGTM

**Pull request closed without merging**